DependencyInjection/Compiler/DoctrineQueryExtensionPass.php

```diff
@@ -40,7 +42,12 @@ public function process(ContainerBuilder $container)
         $extensions = call_user_func_array('array_merge', $sortedServices);
         $dataProviderDefinition = $container->getDefinition('api.doctrine.orm.data_provider');
         foreach ($extensions as $extension) {
-            $dataProviderDefinition->addMethodCall('addExtension', [$extension]);
+            $extensionClass = $container->getDefinition($extension)->getClass();
+            if (is_subclass_of($extensionClass, QueryItemExtensionInterface::class)) {
```

Comment:
I'd say that finding services by tag is more expensive than the `is_subclass_of` call, so I'd leave it this way.